src/TestFramework/Microsoft.Rest.ClientRuntime.Azure.TestFramework/MockContext.cs

```diff
@@ -72,32 +70,59 @@ static MockContext()
         {
             T client;
             handlers = AddHandlers(currentEnvironment, handlers);
+            var constructors = typeof(T).GetConstructors();
 
+            Type tokeCredType = Type.GetType("Microsoft.Rest.TokenCredentials, Microsoft.Rest.ClientRuntime");
```

Comment:
This will break the Search SDK tests when we upgrade to the latest TestFramework. This code should *never* assume that all service clients take `TokenCredentials`. Previously, test setup code for the Search SDK was able to set the `TestFramework.Credentials` property to be a `SearchCredentials` instance. Now this is not possible. How can I create mock `SearchServiceClient` and `SearchIndexClient` instances without copying and tweaking all of `MockContext`?